### PR TITLE
improve parser for certain scenarios

### PR DESCRIPTION
**Pull request by aris-bb**
I encountered some issues with my stories while parsing and fixed them with these changes. Sorry for not being able to come up with a small repro to demonstrate the issue.

### CODE REVIEW

File: src/parser/extract/compiled/nodes.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export async function extractCompiledASTNodes(params: Params): Promise<CompiledA
         id.type === 'ObjectPattern' &&
         init?.type === 'CallExpression' &&
         init.callee.type === 'Identifier' &&
-        init.callee.name === state.defineMetaImport?.local.name
+        init.callee.name === AST_NODES_NAMES.defineMeta
```
Comment (xeho91) [inline src/parser/extract/compiled/nodes.ts:99]: With this change, the user loses the ability to rename the import specifier of `defineMeta` to whatever he prefers.
For example:

```ts
import { defineMeta as dm } from "@storybook/addon-svelte-csf";
```

Why was this change necessary in your case?

File: src/parser/extract/compiled/story.ts [hunk 2]
```diff
@@ -26,8 +26,14 @@ interface Params {
...
-  if (node.type === 'CallExpression' && node.arguments[1].type === 'ObjectExpression') {
-    return node.arguments[1];
+  if (node.type === 'CallExpression') {
+    const secondArg = node.arguments[1];
+    if (secondArg.type === 'ObjectExpression') {
+      return secondArg;
+    }
+    if (secondArg.type === 'CallExpression' && secondArg.arguments[0].type === 'ObjectExpression') {
+      return secondArg.arguments[0];
+    }
```
Comment (xeho91) [inline src/parser/extract/compiled/story.ts:36]: I need to understand why those changes are necessary.

So far I see that it looks like there's a difference in the compiled output between some RC versions of `svelte@5.0.0-next`. Or there's something else I did possibly overlook.

**Review by xeho91 — COMMENTED** [review]
Hi, thanks for contributing and attempting to solve on your own.

Without demonstration for the necessity of those changes is quite difficult for me to understand what sort of issues did you encounter. I suspect there might be more to just those small changes. Could you find a time to give us more insight as to what happened?

File: src/parser/extract/compiled/story.ts (hunk 2 above)
Comment (aris-bb) [reply src/parser/extract/compiled/story.ts:36]: This one stems from a workaround that I had been using before story-level tags support.

I had the following workaround:
```js
// workaround: normally you should use `tags={['!autodocs']}`,
// but `@storybook/addon-svelte-csf`'s StoryProps type does not have a `tags` property,
// thus causing a type error despite working correctly.
const noDocs = () => {
	return {
		tags: ['!autodocs']
	};
};
```

Which I had used like this (first story doesn't use it, and the second one does):
```svelte
<Story
	name="Lifted Targets"
	args={{
		targets: [0x1000, 0x2000, 0x3000],
		manualTargets: [0x4000, 0x5000, 0x6000],
		value: [0x1000, 0x2000, 0x4000, 0x5000],
		liftedTargets: [0x1000, 0x4000]
	}}
	children={template}
/>
<Story
	name="Search Miscellaneous Values"
	args={{ targets: [0x1000, 0x2000], manualTargets: [0x3000, 0x4000] }}
	play={searchMiscellaneousValues}
	{...noDocs()}
	children={template}
/>
```

This results in the following compiled output. You can see that the second `Story(node_6, ...` will not be parsed with the current code:
```js
var node_5 = $.sibling(node_4, 2);

Story(node_5, {
	name: "Lifted Targets",
	args: {
		targets: [0x1000, 0x2000, 0x3000],
		manualTargets: [0x4000, 0x5000, 0x6000],
		value: [0x1000, 0x2000, 0x4000, 0x5000],
		liftedTargets: [0x1000, 0x4000]
	},
	children: template,
	parameters: {
		__svelteCsf: {
			rawCode: "<TargetPicker\n\ttargets={new SvelteSet(args.targets)}\n\tmanualTargets={new SvelteSet(args.manualTargets)}\n\tliftedTargets={new SvelteSet(args.liftedTargets)}\n\tmaxListHeight=\"800px\"\n\tbind:value\n/>"
		}
	}
});

var node_6 = $.sibling(node_5, 2);
var spread_element = $.derived(noDocs);

Story(node_6, $.spread_props({
	name: "Search Miscellaneous Values",
	args: {
		targets: [0x1000, 0x2000],
		manualTargets: [0x3000, 0x4000]
	},
	play: searchMiscellaneousValues,
	parameters: {
		__svelteCsf: {
			rawCode: "<TargetPicker\n\ttargets={new SvelteSet(args.targets)}\n\tmanualTargets={new SvelteSet(args.manualTargets)}\n\tliftedTargets={new SvelteSet(args.liftedTargets)}\n\tmaxListHeight=\"800px\"\n\tbind:value\n/>"
		}
	}
}, () => $.get(spread_element), {
	children: template
}));
```

After the (very!) recent addition of support for story-level tags, if I remove my workaround and use `tags` as usual, the current code is able to parse it correctly.
Therefore, this modification is apparently not needed for my use case.
It might be useful to keep though, in case anybody else is using weird object spread tricks, but I'm not sure if this should be supported or not.

File: src/parser/extract/compiled/nodes.ts (hunk 1 above)
Comment (aris-bb) [reply src/parser/extract/compiled/nodes.ts:99]: For some reason, the variable declaration appears before the import statement for me.
I now realize that the user will be unable to rename the import specifier with my approach.
Unfortunately, I'm not sure how to fix this. Maybe a double-pass approach is required?

Anyway, here is the relevant source snippet:
```svelte
<script lang="ts" context="module">
	import { defineMeta } from '@storybook/addon-svelte-csf';
	import type { Meta } from '@storybook/svelte';
	import { expect, userEvent, within } from '@storybook/test';
	import type { Addon_StoryContext, StepFunction, WebRenderer } from '@storybook/types';
	import { queries } from '@testing-library/dom';

	import TargetPicker from './TargetPicker.svelte';

	const { Story } = defineMeta({
		title: 'App/Workspace/Graph View/Node/Partials/Target Picker',
		component: TargetPicker,
		tags: ['autodocs']
	} satisfies Meta<TargetPicker>);
</script>
```

Here is the compiled output that I get:
```js
import {createHotContext as __vite__createHotContext} from "/@vite/client";
import.meta.hot = __vite__createHotContext("/src/lib/components/app/workspace/graph-view/node/target-picker/TargetPicker.stories.svelte");
import "/node_modules/.cache/storybook/<hash>/sb-vite/deps/svelte_internal_disclose-version.js?v=f43de63d";

$.mark_module_start();
TargetPicker_stories[$.FILENAME] = "src/lib/components/app/workspace/graph-view/node/target-picker/TargetPicker.stories.svelte";

const {Story, meta} = defineMeta({
    title: 'App/Workspace/Graph View/Node/Partials/Target Picker',
    component: TargetPicker,
    tags: ['autodocs']
});

import*as $ from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/svelte_internal_client.js?v=f43de63d";
import {defineMeta} from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/@storybook_addon-svelte-csf.js?v=f43de63d";
import {expect, userEvent, within} from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/@storybook_test.js?v=f43de63d";
import {queries} from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/@testing-library_dom.js?v=f43de63d";
import TargetPicker from "/src/lib/components/app/workspace/graph-view/node/target-picker/TargetPicker.svelte";
```

For completeness' sake, here's the compiled output I get if I `import { defineMeta as dm }`:
```js
import {createHotContext as __vite__createHotContext} from "/@vite/client";
import.meta.hot = __vite__createHotContext("/src/lib/components/app/workspace/graph-view/node/target-picker/TargetPicker.stories.svelte");
import "/node_modules/.cache/storybook/<hash>/sb-vite/deps/svelte_internal_disclose-version.js?v=f43de63d";

$.mark_module_start();
TargetPicker_stories[$.FILENAME] = "src/lib/components/app/workspace/graph-view/node/target-picker/TargetPicker.stories.svelte";

const {Story, meta} = dm({
    title: 'App/Workspace/Graph View/Node/Partials/Target Picker',
    component: TargetPicker,
    tags: ['autodocs']
});

import*as $ from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/svelte_internal_client.js?v=f43de63d";
import {defineMeta as dm} from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/@storybook_addon-svelte-csf.js?v=f43de63d";
import {expect, userEvent, within} from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/@storybook_test.js?v=f43de63d";
import {queries} from "/node_modules/.cache/storybook/<hash>/sb-vite/deps/@testing-library_dom.js?v=f43de63d";
import TargetPicker from "/src/lib/components/app/workspace/graph-view/node/target-picker/TargetPicker.svelte";
```

Comment (aris-bb) [reply src/parser/extract/compiled/nodes.ts:99]: UPDATE: I just found out about [this](https://github.com/sveltejs/svelte/issues/13135), which is the root of the problem. Updating to the newest Svelte 5 version renders this change unnecessary.

**Comment by aris-bb** [comment]
After digging deeper to answer the questions, I realized these changes were not necessary in the end. Therefore I'm closing this. Thanks for the review, and sorry for the hassle!

File: src/parser/extract/compiled/story.ts (hunk 2 above)
Comment (xeho91) [reply src/parser/extract/compiled/story.ts:36]: > This one stems from a workaround that I had been using before story-level tags support.
> 
> I had the following workaround:
> 
> ```js
> // workaround: normally you should use `tags={['!autodocs']}`,
> // but `@storybook/addon-svelte-csf`'s StoryProps type does not have a `tags` property,
> // thus causing a type error despite working correctly.
> const noDocs = () => {
> 	return {
> 		tags: ['!autodocs']
> 	};
> };
> ```

FYI, this one for sure wouldn't work correctly. This addon does parsing firstly for Storybook indexer the following `defineMeta` properties:
- `title`
- `tags`

and the following `<Story />` props:
- `name`
- `exportName`
- `tags`

Meaning that those attributes cannot be dynamic, nor have any linked references. Meaning they have to be static literal string or array of static literal strings _(for `tags`)_.

**Comment by xeho91** [comment]
**Thank you very much** for the thorough explanations and exploration! And for trying out this addon
